src/gui/transferlistsortmodel.cpp

```diff
@@ -219,16 +233,14 @@ bool TransferListSortModel::lessThan(const QModelIndex &left, const QModelIndex
 {
     Q_ASSERT(left.column() == right.column());
 
-    if (m_lastSortColumn != left.column())
-    {
-        if (m_lastSortColumn != -1)
-            m_subSortColumn = m_lastSortColumn;
-        m_lastSortColumn = left.column();
-    }
-
     const int result = compare(left, right);
-    if (result == 0)
-        return compare(left.sibling(left.row(), m_subSortColumn), right.sibling(right.row(), m_subSortColumn)) < 0;
+    if (result == 0) {
+        const int subResult = compare(left.sibling(left.row(), m_subSortColumn), right.sibling(right.row(), m_subSortColumn));
+        const bool orderDescending = (m_subSortColumnOrder == 1);
```

Comment:
I think it is better that you use `orderAscending` as in the code in `TransferListSortModel::sort()`.